### DISCUSSION TITLE
Templating Question

### DISCUSSION THREAD

**Discussion by jtenner** [parcel-bundler/parcel#9702] [Q&A] [answered]
I use parcel at work all the time to deliver a very important mission critical static application every day, and for that I am truly thankful.

However, I have another potential use case that is giving me trouble, and I'm having trouble understanding how to use plugins properly for my needs.

Let's say I have the following file structure in my static site, and use the following command.

```sh
npx parcel src/**/*.*
```
This is the file layout.

```
src/_layout.ejs
src/_layout.yaml
src/index.md
```
I would like to produce an HTML asset for each markdown file (and also each ejs file that doesn't start with an "_".) When the md file is rendered, I want to use the _layout.ejs file in the directory of the markdown, and then nest that rendered content into each parent directory's layout file.

Given the following layout:

```
src/_layout.yaml
src/_layout.ejs
src/subfolder/_layout.ejs
src/subfolder/_layout.yaml
src/subfolder/index.md
```

The following things would happen:

1. render `index.md` as github flavored markdown
2. render the html *inside* `src/subfolder/_layout.ejs` with data `mixin(rootYaml, subfolderYaml, indexFrontmatter)`
3. render the html created above in `src/_layout.ejs` with the same data

Of course, the HTML files would then function like normal.

Does anyone have any suggestions on how to achieve this?

**Comment by mischnic** [comment]
What you need is something like https://github.com/mischnic/parcel-ssg

If you really only need "dumb" templating, then you really only need a transformer that converts markdown to HTML (and respects the layout files in parent folders).

If you need more of the features usually provided by e.g. Eleventy ("global" information like generating a navbar with list of pages, RSS feed), then it becomes more complicated

**Reply by jtenner** [reply]
Thank you for the very quick reply! I have a very custom version of EJS templates because EJS hasn't been maintained in a few years, so finding a plug-able SSG has been very challenging.

It looks like this *could* fit my needs. Are there any other alternatives?

For instance, if I simply tell parcel to look at everything in the `./src` directory, is it possible to generate a "bundle" for each markdown file, and find each asset that would be logically grouped with each generated page?

I have no understanding of how most of the plugins work, and I'm not sure where to look.

**Reply by mischnic** [reply, marked as answer]
1. You run `parcel build 'src/**/*.md'`
2. Then there would be a transformer plugin that just translates markdown into the equivalent HTML 
3. Then it behaves just like using HTML source files with Parcel (the assets that are referenced in HTML are automatically detected and processed as well).

Does that answer your questions?

**Reply by jtenner** [reply]
Yeah of course! The only issue I see is that every markdown file would be templated by a file in the root directory, and I can't do relative urls. I think this will have to do.
